source/MaterialXGenMdl/mdl/materialx/noise.mdl

```diff
@@ -482,6 +482,77 @@ export float mx_fractal_noise_float(
     return result;
 }
 
+export float3 mx_fractal2d_noise_float3(
+  float2 mxp_p = core::mx_swizzle_xy( ::state::texture_coordinate(0)))
+  int mxp_octaves = 3,
+  float mxp_lacunarity = 2.0,
+  float mxp_diminish= 0.5)
+[[
+    anno::noinline()
+]]
+{
+    float3 result = float3(0.0);
+    float amplitude = 1.0;
+    for (int i = 0;  i < mxp_octaves; ++i)
+    {
+        result += amplitude * mx_perlin_noise_float3(mxp_p);
+        amplitude *= mxp_diminish;
+        mxp_p *= mxp_lacunarity;
+    }
+    return result;
+}
+
+export float2 mx_fractal2d_noise_float2(
+  float2 mxp_p = core::mx_swizzle_xy( ::state::texture_coordinate(0)))
+  int mxp_octaves = 3,
+  float mxp_lacunarity = 2.0,
+  float mxp_diminish= 0.5)
+[[
+    anno::noinline()
+]]
+{
+    return float2(mx_fractal2d_noise_float(mxp_p, mxp_octaves, mxp_lacunarity, mxp_diminish),
+                  mx_fractal2d_noise_float(mxp_p+float2(19, 193), mxp_octaves, mxp_lacunarity, mxp_diminish));
+}
+
+export float4 mx_fractal2d_noise_float4(
+  float2 mxp_p = core::mx_swizzle_xy( ::state::texture_coordinate(0)))
+  int mxp_octaves = 3,
+  float mxp_lacunarity = 2.0,
+  float mxp_diminish= 0.5)
+[[
+    anno::noinline()
+]]
+{
+
+    float3 c = mx_fractal2d_noise_float3(mxp_p, mxp_octaves, mxp_lacunarity, mxp_diminish);
+    float a = mx_fractal2d_noise_float(mxp_p+float2(19, 193), mxp_octaves, mxp_lacunarity, mxp_diminish);
+    return float4(c.x, c.y, c.z, a);
+}
+
+// TODO - do we want to rename these to mx_fractal3D_noise_float - if so where else might things need to be updated
```

Comment:
That rename sounds logical to me, and I'm CC'ing @krohmerNV for his thoughts from the MDL side.